src/Hazelcast.Net/Networking/SslLayer.cs

```diff
@@ -190,5 +188,17 @@ internal bool ValidateCertificate(object sender, X509Certificate cert, X509Chain
 
             return validation;
         }
+
+        // Methodized for testing. 
+        public string GetTargetHostNameOrDefault()
+        {
+            // if targetHost does not match the server certificate name then a RemoteCertificateNameMismatch error will
+            // be reported, which can be ignored with options.ValidateCertificateName being false. If it is true, then
+            // options.CertificateName *must* be set to the server certificate name.
+            // + if no options.CertificateName is given, .Net 4.8 assigns random numbers which fails on Hazelcast at Java 17+.
+            // We set the target host name to connecting address if nothing to have. See issue #800.
+
+            return _options.CertificateName ?? _ipAddress.ToString();
```

Comment:
can u add a //TODO comment saying we should refactor this and use a new _options.RemoteName or whatever?

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/812/commits/334a80863b25f869cc68077a4e41ebaec9fe1f33